libimage/runtime.go

```diff
@@ -233,13 +239,29 @@ func (r *Runtime) LookupImage(name string, options *LookupImageOptions) (*Image,
 		name = normalizedName
 	}
 
+	byDigest := false
 	originalName := name
-	idByDigest := false
 	if strings.HasPrefix(name, "sha256:") {
-		// Strip off the sha256 prefix so it can be parsed later on.
-		idByDigest = true
+		byDigest = true
 		name = strings.TrimPrefix(name, "sha256:")
 	}
+	byFullID := regexFullID.MatchString(name)
+
+	if byDigest && !byFullID {
```

Comment:
FWIW a cleaner (but technically less efficient) approach would be, in the `if` above,
```go
d, err := digest.Parse(name);
if err != nil { fail() }
name = d.Encoded()
```
and then the `byFullID` check can be only done in the non-sha256-prefixed case.

Reply:
Has now been replaced with the new `reference.IsFullIdentifier()`.